src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs

```diff
@@ -216,7 +216,7 @@ private static bool TryGetTextForOperator(SyntaxToken token, Document document,
             }
 
             var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            if (syntaxFacts.IsOperator(token) || syntaxFacts.IsPredefinedOperator(token) || SyntaxFacts.IsAssignmentExpressionOperatorToken(token.Kind()))
+            if (syntaxFacts.IsOperator(token) || SyntaxFacts.IsAssignmentExpressionOperatorToken(token.Kind()))
```

Comment:
"IsPredefinedOperator" only checks the token kind regardless its context. This caused `<` of type argument list (e.g, in `List<int>`) to be classified as less than.

The check of "IsOperator" should handle everything with respect to the token context. I also think `IsAssignmentExpressionOperatorToken` should be removed, but will leave that for now because a test will fail, and F1 isn't the main change intended with this PR.

Reply:
I wouldn't touch this function at all in this pr. It seems to be fixing two unrelated issues.

Reply:
@CyrusNajmabadi I had to touch it to fix test failure.

Reply:
the change feels dangerous.  i'm ont sure what unintended consequences tehre would be by removing this case here.  Instead of removing htis, could we instead have the C# f1 helper check if this a < in a generic name and not in a regular expression?

Reply:
That way I'll have to special case multiple things:

- TypeArgumentListSyntax
- TypeParameterListSyntax
- FunctionPointerParameterListSyntax
- XmlElementStartTagSyntax
- OperatorMemberCref
- XmlEmptyElementSyntax

This doesn't look neat IMO.

Either way, F1 help service isn't accurate and has bugs.

Does adding more tests to cover all operators that are in the deleted checks make the change more comfortable?

Reply:
or just say that `<` is only an operator here if it's parent is a binaryexpression